conbench/app/_plots.py

```diff
@@ -112,7 +112,8 @@ def get_history_plot(
                     "you asked to highlight the benchmark result with ID "
                     f"<kbd>{highlight_other_result.bmrid}</kbd> in this view. "
                     "However, that result is not part of the history of the "
-                    f"result <kbd>{current_benchmark_result.id}</kbd>",
+                    f"result <kbd>{current_benchmark_result.id}</kbd>. The "
+                    "two results are not directly comparable.",
```

Comment:
I think this needs more nuance. 

There are reasons for making this kind of comparison, e.g. when Arrow devs want to compare a release candidate on a branch to a previous release candidate on another branch. They're not directly connected by git history so we can't do z-score analysis, but pairwise analysis might still be interesting.

I'm not sure what wording best captures the nuance, but maybe just specifying what specific analyses can and can't be done.